### DISCUSSION TITLE
Allow access to users if they are in same user_group.

### DISCUSSION THREAD

**Discussion by neerajnigam6** [pocketbase/pocketbase#4146] [Q&A] [answered]
I want to allow users to read other users only if they are part of same gorup.
something like -
```SELECT * FROM User u JOIN user_group ug ON ug.user = u.id JOIN group g ON ug.group = g.id WHERE g.id IN (SELECT id FROM group JOIN user_group WHERE user_group.user = ${@request.auth.id}) ```

However I am not sure how can I translate the same to API rules (or do it programmatically via go)

Here is my collection structure:
1. User (default user collection)
2. Group (custom collection):
  id: string,
  name: string
4. user_group:
  id: string,
  user: user_id (foreign key)
  group: group_id (foreign key)

**Comment by ganigeorgiev** [comment, marked as answer]
Your use case seems similar to https://github.com/pocketbase/pocketbase/discussions/3805.

If I'm understanding your requirements, you can try with the following rule:

```js
@request.auth.id != "" &&
// join the user_group for the non-auth record
@collection.user_group.user ?= id &&
// join the user_group for the auth record
@collection.user_group:auth.user ?= @request.auth.id &&
// make sure that the the auth group matches with the user one
@collection.user_group:auth.group ?= @collection.user_group.group
```

Note that the `@collection.*:alias` was added in PocketBase v0.20.0. The above eventually will be further simplified once we add support for back/indirect relations.

**Reply by neerajnigam6** [reply]
Thanks, it worked
